internal/defragger/defrag.go

```diff
@@ -0,0 +1,45 @@
+package defragger
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/improbable-eng/etcd-cluster-operator/internal/etcd"
+)
+
+type UsedSpaceChecker interface {
+	UsedSpacePercentage(context.Context, etcd.Member) (int, error)
+}
+
+type Defragger interface {
+	Defragment(context.Context, etcd.Member) error
+}
+
+// DefragIfNecessary will defrag each etcd member if necessary, it will return the last error encountered
+func DefragIfNecessary(ctx context.Context, memberList []etcd.Member, s UsedSpaceChecker, d Defragger, log logr.Logger) error {
+	var returnErr error
+	for _, member := range memberList {
+		usedSpace, err := s.UsedSpacePercentage(ctx, member)
+		if err != nil {
+			log.Error(err, "unable to check if defrag is necessary")
+			returnErr = err
+			continue
+		}
+
+		if usedSpace > 80 {
```

Comment:
Does it make sense to make this configurable?

Reply:
Also i think a swap and early continue should make code readable.

Reply:
Done :)